src/shogun/machine/Machine.cpp

```diff
@@ -55,7 +55,29 @@ bool CMachine::train(CFeatures* data)
 	}
 
 	auto sub = connect_to_signal_handler();
-	bool result = train_machine(data);
+	bool result = false;
+
+	if (support_feature_dispatching())
+	{
+		REQUIRE(data != NULL, "Features not provided!");
+		REQUIRE(
```

Comment:
i think the check here is okay ? https://github.com/shogun-toolbox/shogun/pull/4373/files#diff-cc285e900e1599bd903dbb6d435f6089L52

Reply:
the link doesnt work, where is this?

Reply:
[Machine.cpp line 51](https://github.com/shubham808/shogun/blob/c498f80caafb2e421dfd05f88759d1029fd9bb52/src/shogun/machine/Machine.cpp#L51)

Reply:
this is not enough as it is only checked if `training_requires_labels()` returns true. Your code here can segfault if no labels are provided....you need to enforce labels if dispatching is enabled ...
Since we will enforce labels to be passed soon, you can do that for the case of dispatching